src/Controllers/NotFoundRedirectController.cs

```diff
@@ -24,11 +23,21 @@ namespace BVNetwork.NotFound.Controllers
     [Authorize]
     public class NotFoundRedirectController : Controller
     {
-
+        private readonly IRedirectsService _redirectsService;
+        private readonly DdsRedirectRepository _ddsRedirectRepository;
         private static readonly ILogger Logger = LogManager.GetLogger();
+
+        public NotFoundRedirectController(
+            DefaultRedirectsService redirectsService,
+            DdsRedirectRepository ddsRedirectRepository)
+        {
+            _redirectsService = redirectsService ?? throw new ArgumentNullException(nameof(redirectsService));
+            _ddsRedirectRepository = ddsRedirectRepository ?? throw new ArgumentNullException(nameof(ddsRedirectRepository));
+        }
+
         private void CheckAccess()
         {
-            if (!PrincipalInfo.HasEditAccess)
+            if (!EPiServer.Security.PrincipalInfo.HasEditAccess)
```

Comment:
do we need full namespace here?

Reply:
Yes, we need. Episerver.Security namespace contains some extensions which break Linq in other places.